sql/core/src/main/scala/org/apache/spark/sql/execution/ExistingRDD.scala

```diff
@@ -319,3 +319,46 @@ case class RDDScanExec(
 
   override def getStream: Option[SparkDataStream] = stream
 }
+
+/**
+ * A physical plan node for `OneRowRelation` for scans with no 'FROM' clause.
+ *
+ * We do not extend `RDDScanExec` in order to avoid complexity due to `TreeNode.makeCopy` and
+ * `TreeNode`'s general use of reflection.
+ */
+case class OneRowRelationExec() extends LeafExecNode
+  with InputRDDCodegen {
+
+  override val nodeName: String = s"Scan OneRowRelation"
+
+  override val output: Seq[Attribute] = Nil
+
+  private val emptyRow: InternalRow = InternalRow.empty
+
+  private val rdd = session.sparkContext.parallelize(Seq(emptyRow), 1)
```

Comment:
I think we can do this
```
private val rdd = {
  val proj = UnsafeProjection.create(schema)
  val emptyRow = proj(InternalRow.empty)
  session.sparkContext.parallelize(Seq(emptyRow), 1)
}
```

Reply:
then `def doExecute()` can just return this RDD

Reply:
hmm, sure i implemented this - but because we still want to increment the number of output rows at the time when the row is actually processed, i did not end up simply returning `rdd` from `doExecute()`... lmk if you think there a better way.